src/Components/HeroSection.js

```diff
@@ -0,0 +1,61 @@
+import React, {useEffect} from 'react'
+import {Switch, Route} from "react-router-dom"
+import CartList from '../Pages/CartList'
+import ProductsList from '../Pages/ProductsList'
+import ProductDetail from '../Pages/ProductDetail'
+import WishList from '../WishList/WishList'
+import Home from '../Pages/Home'
+import axios from "axios"
+import { useStore } from '../Store/context'
+
+
+const HeroSection = () => {
+    const {dispatch} = useStore()
+    useEffect(() => {
+        async function fetchData() {
+            dispatch({type: "IS_LOADING", payload: "loading"})
+            try {
+                const response = await axios.get("/api/products")
+                const data = await response.data.products
+                if(response.status === 200){
+                    dispatch({type: "IS_LOADING"})
+                    dispatch({type: "LOAD_PRODUCTS", payload: data})
+                }
+            } catch (error) {
+                console.log(error)
+            }
+            finally{
+                dispatch({type: "IS_LOADING", payload: "success"})
+            }
+        }
+    fetchData();
+    },[])
+    useEffect(() => {
+        async function fetchData() {
+            dispatch({type: "IS_LOADING", payload: "loading"})
+            try {
+                const response = await axios.get("/api/cart")
+                const data = await response.data.carts
+                if(response.status){
+                    dispatch({type: "LOAD_CART_ITEMS", payload: data})
+                }
```

Comment:
How about we group both GET requests(for products and cart) in one useEffect?
Let me know what u think!